File: src/main/java/DahRealPanda/plugins/colonyweb/colony/model/CombatInfo.java
package DahRealPanda.plugins.colonyweb.colony.model;

import java.util.ArrayList;
import java.util.List;

/**
 * Colony defence: raid pressure, the guard roster and the buildings backing them.
 */
public class CombatInfo {
    public boolean raidsPossible;
    public boolean underAttack;
    public int nightsSinceRaid;
    public int raidLevel;        // MineColonies' colony raid level (raid difficulty scaling)
    public boolean spiesEnabled;

    public int guardCount;
    public int guardCapacity;    // total guard slots across guard buildings
    public double averageGuardLevel;
    public double averageHealthPct;
    public int graves;           // unclaimed graves — citizens that died and need burying

    public List<Guard> guards = new ArrayList<>();
    public List<Post> posts = new ArrayList<>();
    public List<Event> events = new ArrayList<>();

    /** A citizen with a combat job. */
    public static class Guard {
        public int id;
        public String name;
        public String job;
        public String jobType;
        public int level;        // job-relevant skill level
        public double health;
        public double maxHealth;
        public boolean spawned;
        public String building;
        public int x;
        public int y;
        public int z;
    }

    /** A guard tower / barracks and how well it is staffed. */
    public static class Post {
        public int id;
        public String name;
        public String type;
        public String blockId;
        public int level;
        public int assigned;
        public int capacity;
        public int x;
        public int y;
        public int z;
    }

    /** An active colony event — usually an ongoing raid. */
    public static class Event {
        public int id;
        public String name;
        public String status;
        public int x;
        public int y;
        public int z;
    }
}

File: common/src/main/kotlin/DahRealPanda/plugins/colonyweb/model/CombatInfo.kt
package DahRealPanda.plugins.colonyweb.model

/**
 * Colony defence: raid pressure, the guard roster and the buildings backing them.
 */
data class CombatInfo(
    @JvmField var raidsPossible: Boolean = false,
    @JvmField var underAttack: Boolean = false,
    @JvmField var nightsSinceRaid: Int = 0,
    @JvmField var raidLevel: Int = 0,        // MineColonies' colony raid level (raid difficulty scaling)
    @JvmField var spiesEnabled: Boolean = false,
    @JvmField var guardCount: Int = 0,
    @JvmField var guardCapacity: Int = 0,    // total guard slots across guard buildings
    @JvmField var averageGuardLevel: Double = 0.0,
    @JvmField var averageHealthPct: Double = 0.0,
    @JvmField var graves: Int = 0,           // unclaimed graves — citizens that died and need burying
    @JvmField var guards: MutableList<Guard> = ArrayList(),
    @JvmField var posts: MutableList<Post> = ArrayList(),
    @JvmField var events: MutableList<Event> = ArrayList()
) {
    /** A citizen with a combat job, and the kit they are carrying. */
    data class Guard(
        @JvmField var id: Int = 0,
        @JvmField var name: String = "",
        @JvmField var job: String = "",
        @JvmField var jobType: String = "",
        @JvmField var level: Int = 0,        // job-relevant skill level
        @JvmField var health: Double = 0.0,
        @JvmField var maxHealth: Double = 0.0,
        @JvmField var spawned: Boolean = false,
        @JvmField var building: String = "",      // the guard post they are stationed at
        @JvmField var buildingId: Int = -1,
        @JvmField var buildingLevel: Int = 0,    // that post's level
        @JvmField var equipment: MutableList<EquipmentInfo> = ArrayList(),
        @JvmField var armorPoints: Int = 0,      // total vanilla armour value across the four slots
        @JvmField var weapon: String? = null,     // main-hand item name, null when empty-handed
        @JvmField var x: Int = 0,
        @JvmField var y: Int = 0,
        @JvmField var z: Int = 0
    )

    /** A guard tower / barracks and how well it is staffed. */
    data class Post(
        @JvmField var id: Int = 0,
        @JvmField var name: String = "",
        @JvmField var type: String = "",
        @JvmField var blockId: String = "",
        @JvmField var level: Int = 0,
        @JvmField var assigned: Int = 0,
        @JvmField var capacity: Int = 0,
        @JvmField var x: Int = 0,
        @JvmField var y: Int = 0,
        @JvmField var z: Int = 0
    )

    /** An active colony event — usually an ongoing raid. */
    data class Event(
        @JvmField var id: Int = 0,
        @JvmField var name: String = "",
        @JvmField var status: String = "",
        @JvmField var x: Int = 0,
        @JvmField var y: Int = 0,
        @JvmField var z: Int = 0
    )
}
